src/components/Extension/CollectDataExtension/actions/RemoteSingleSelectAction/RemoteSingleSelectAction.tsx

```diff
@@ -0,0 +1,143 @@
+import React, { FC, useEffect, useMemo, useState } from 'react'
+import { mapActionFieldsToObject } from '../../../utils'
+import classes from './remoteSingleSelectAction.module.css'
+
+import type { RemoteSingleSelectActionFields } from '../../types'
+import type { ExtensionActivityRecord } from '../../../types'
+import { useRemoteSingleSelectAction } from './hooks/useRemoteSingleSelectAction'
+import { Button, Select, type Option } from '@awell-health/ui-library'
+import { useTranslation } from 'next-i18next'
+import { isNil, debounce } from 'lodash'
+
+interface RemoteSingleSelectActionProps {
+  activityDetails: ExtensionActivityRecord
+}
+
+export const RemoteSingleSelectAction: FC<RemoteSingleSelectActionProps> = ({
+  activityDetails,
+}) => {
+  const { activity_id, fields } = activityDetails
+  const { t } = useTranslation()
+  const { onSubmit, isSubmitting } = useRemoteSingleSelectAction()
+  const [selectedOption, setSelectedOption] = useState<Option>()
+  const [options, setOptions] = useState<Array<Option>>([])
+  const [error, setError] = useState<unknown>()
+  const [searchText, setSearchText] = useState<string>('')
+
+  const handleFetchOptions = async (
+    url: string,
+    queryParam: string,
+    headers: string,
+    search = '',
+    onError?: (error: unknown) => void
+  ) => {
+    try {
+      const response = await fetch(
+        `${url}/${queryParam && search ? '?' + queryParam + '=' + search : ''}`,
+        {
+          method: 'GET',
+          headers: {
+            'Content-Type': 'application/json',
+            'ngrok-skip-browser-warning': 'yes',
+            ...JSON.parse(headers),
+          },
+        }
+      )
+      const options = await response.json()
+      return options
+    } catch (error) {
+      console.error(error)
+      if (!isNil(onError)) {
+        onError(error)
+      }
+    }
+  }
+
+  const {
+    queryParam,
+    url,
+    label,
+    headers = '{}',
+    mandatory = false,
+  } = useMemo(
+    () => mapActionFieldsToObject<RemoteSingleSelectActionFields>(fields),
+    [fields]
+  )
+
+  const fetchOptions = debounce(async () => {
+    const options = await handleFetchOptions(
+      url,
+      queryParam,
+      headers,
+      searchText,
+      setError
+    )
+    setOptions(options)
+  }, 1000)
+
+  const handleSubmit = () => {
+    if (isNil(selectedOption) || isNil(selectedOption.value)) {
+      setError(t('activities.form.question_required_error'))
+      return
+    }
+    onSubmit({
+      activityId: activity_id,
+      label: selectedOption.label,
+      value: selectedOption.value.toString(),
+    })
+  }
+
+  console.log(mandatory)
```

Comment:
Can we remove the console logs?
They are also in Extension.tsx line 25 & completeFlowAction line 44 (introduced 9 months ago)